core/tests/protractor_utils/users.js

```diff
@@ -43,13 +43,18 @@ var _createFirebaseAccount = async function(email, isSuperAdmin = false) {
   }
 };
 
-var login = async function(email) {
-  // Use of element and action is not reliable, because we do not always begin
-  // on an Angular page. To forgive callers from non-Angular pages (e.g., the
-  // very first function call of a test), we use browser.driver instead.
-  // The full url is necessary.
-  await browser.driver.get(
-    general.SERVER_URL_PREFIX + general.LOGIN_URL_SUFFIX);
+var login = async function(email, useLoginButton = false) {
+  if (useLoginButton) {
+    var loginButton = element(by.css('.protractor-test-login-button'));
+    await action.click('Login button', loginButton);
+  } else {
```

Comment:
It might be worth explaining under what circumstances the "else" flow is triggered (since people may wonder: how else would you log in without using a login button?).

I'm also a bit concerned about adding a new parameter to each API function just because of one use case. Can we avoid that? That's typically an antipattern and there's usually some way of doing this that doesn't increase complexity.

Reply:
OK, I've decided to inline the `loginButton` action into the test, and bring back `login`'s `manualNavigation` argument. 
This way the argument acts as a simple opt-out flag.

Reply:
Looks good, thanks for changing that and not propagating the extra arg all over the API.

If you don't mind, would it be OK to add some documentation explaining what manualNavigation is and when to use it? I know this predates your PR, so it's optional, but I think it would be helpful.

Reply:
Isn't that already done in the function itself? What's missing from the comment?

Reply:
It doesn't tell future developers when manualNavigation should be used (or at least, it's not clear). It would help to write it as something like the following:

```
Note that callers of this function should only set `manualNavigation` to true when XXX, and false otherwise. This  is necessary because YYY.
```

Reply:
OK, done.